anymail/backends/mailtrap.py

```diff
@@ -0,0 +1,272 @@
+
+    def set_esp_extra(self, extra):
+        update_deep(self.data, extra)
+
+
+class EmailBackend(AnymailRequestsBackend):
+    """
+    Mailtrap API Email Backend
+    """
+
+    esp_name = "Mailtrap"
+
+    def __init__(self, **kwargs):
+        """Init options from Django settings"""
+        self.api_token = get_anymail_setting(
+            "api_token", esp_name=self.esp_name, kwargs=kwargs, allow_bare=True
+        )
+        api_url = get_anymail_setting(
+            "api_url",
+            esp_name=self.esp_name,
+            kwargs=kwargs,
+            default="https://send.api.mailtrap.io/api/",
+        )
+        if not api_url.endswith("/"):
+            api_url += "/"
+
+        test_api_url = get_anymail_setting(
+            "test_api_url",
+            esp_name=self.esp_name,
+            kwargs=kwargs,
+            default="https://sandbox.api.mailtrap.io/api/",
+        )
+        if not test_api_url.endswith("/"):
+            test_api_url += "/"
+        self.test_api_url = test_api_url
+
+        self.testing_enabled = get_anymail_setting(
+            "testing",
+            esp_name=self.esp_name,
+            kwargs=kwargs,
+            default=False,
+        )
+
+        if self.testing_enabled:
+            self.test_inbox_id = get_anymail_setting(
+                "test_inbox_id",
+                esp_name=self.esp_name,
+                kwargs=kwargs,
+                # (no default means required -- error if not set)
+            )
+            api_url = self.test_api_url
+        else:
+            self.test_inbox_id = None
+
+        super().__init__(api_url, **kwargs)
+
+    def build_message_payload(self, message, defaults):
+        return MailtrapPayload(message, defaults, self)
+
+    def parse_recipient_status(
+        self, response, payload: MailtrapPayload, message: AnymailMessage
+    ):
+        parsed_response = self.deserialize_json_response(response, payload, message)
+
+        # TODO: how to handle fail_silently?
+        if not self.fail_silently and (
+            not parsed_response.get("success")
+            or ("errors" in parsed_response and parsed_response["errors"])
+            or ("message_ids" not in parsed_response)
+        ):
+            raise AnymailRequestsAPIError(
+                email_message=message, payload=payload, response=response, backend=self
+            )
+        else:
+            # message-ids will be in this order
+            recipient_status_order = [
+                *payload.recipients_to,
+                *payload.recipients_cc,
+                *payload.recipients_bcc,
+            ]
+            recipient_status = {
+                email: AnymailRecipientStatus(
+                    message_id=message_id,
+                    status="sent",
+                )
+                for email, message_id in zip(
+                    recipient_status_order, parsed_response["message_ids"]
+                )
+            }
```

Comment:
Here's an approach for reporting invalid responses. It's also probably worth checking the number of message_ids, since zip() won't raise an error for that. 

```suggestion
        try:
            message_ids = parsed_response["message_ids"]
        except KeyError as err:
            raise AnymailRequestsAPIError(
                "Invalid response format: missing message_ids",
                email_message=message, payload=payload, response=response, backend=self
            ) from err

        # message-ids will be in this order
        recipient_status_order = [
            *payload.recipients_to,
            *payload.recipients_cc,
            *payload.recipients_bcc,
        ]
        if len(recipient_status_order) != len(message_ids):
            raise AnymailRequestsAPIError(
                "Invalid response format: wrong number of message_ids",
                email_message=message, payload=payload, response=response, backend=self
            )
            
        recipient_status = {
            email: AnymailRecipientStatus(
                message_id=message_id,
                status="sent",
            )
            for email, message_id in zip(recipient_status_order, message_ids)
        }
```

API error responses should already be covered by raise_for_status() in the superclass. (But if you wanted to add a check like `if parsed_response.get("errors") or not parsed_response.get("success"): raise AnymailRequestsAPIError("Unexpected errors in {response.status_code} response", ...)`, that wouldn't hurt anything.)